arm/Microsoft.Storage/storageAccounts/deploy.bicep

```diff
@@ -165,7 +172,8 @@ var saBaseProperties = {
   isHnsEnabled: ((!enableHierarchicalNamespace) ? null : enableHierarchicalNamespace)
   minimumTlsVersion: minimumTlsVersion
   networkAcls: (empty(networkAcls) ? null : networkAcls_var)
-  allowBlobPublicAccess: allowBlobPublicAccess
+  allowBlobPublicAccess: (storageAccountKind == 'true') ? null : allowBlobPublicAccess
```

Comment:
Change to
```suggestion
  allowBlobPublicAccess: allowBlobPublicAccess == 'true' ? allowBlobPublicAccess : null
```

Reply:
Agree to the swap around, its just to drive consistency on having the change being the "true" approach, while the "false" resulting in no change, being `null`